ntpd/src/daemon/keyexchange.rs

```diff
@@ -647,4 +660,243 @@ mod tests {
             _ => panic!(),
         }
     }
+
+    fn client_key_exchange_message_length() -> usize {
+        let mut buffer = Vec::with_capacity(1024);
+        for record in ntp_proto::NtsRecord::client_key_exchange_records() {
+            record.write(&mut buffer).unwrap();
+        }
+
+        buffer.len()
+    }
+
+    async fn send_records_to_client(
+        records: Vec<NtsRecord>,
+    ) -> Result<KeyExchangeResult, KeyExchangeError> {
+        let listener = tokio::net::TcpListener::bind(("localhost", 0))
+            .await
+            .unwrap();
+        let port = listener.local_addr()?.port();
```

Comment:
using port 0 makes the OS assign a unique port to this stream. We just need to make sure it is used consistently